.github/workflows/docker-image.yml

```diff
@@ -0,0 +1,86 @@
+name: Docker Image CI
+
+on:
+  push:
+    tags:
+      - '*' 
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME_DAPP: ${{ github.repository }}-dapp
+  IMAGE_NAME_PUBLISHER: ${{ github.repository }}-publisher
+
+jobs:
+  build-and-push-dapp:
+    runs-on: ubuntu-latest
+
+    permissions:
+      contents: read
+      packages: write
+
+    steps:
+    - name: Check out the repo
+      uses: actions/checkout@v4
+
+    - name: Log in to the Container registry
+      uses: docker/login-action@v3
+      with:
+        registry: ${{ env.REGISTRY }}
+        username: ${{ github.actor }}
+        password: ${{ secrets.GITHUB_TOKEN }}
+
+    - name: Extract metadata (tags, labels) for Docker
+      id: meta
+      uses: docker/metadata-action@v5
+      with:
+        images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME_DAPP }}
+        # Set the list of tags for the Docker image
+        tags: |
+          type=ref,event=tag
+          type=raw,value=latest
+
+    - name: Build and push Docker image
+      uses: docker/build-push-action@v5
+      with:
+        context: ./dapp
+        file: ./dapp/docker/Dockerfile
+        push: true
+        tags: ${{ steps.meta.outputs.tags }}
+        labels: ${{ steps.meta.outputs.labels }}
+
+  build-and-push-publisher:
+    runs-on: ubuntu-latest
```

Comment:
Same as above

Reply:
Kind of recommended by Github in all of their examples. No particular reason, we can use alpine as well. https://docs.github.com/en/actions/publishing-packages/publishing-docker-images#publishing-images-to-github-packages